libkv/libkv.go

```diff
@@ -0,0 +1,258 @@
+package libkv
+
+import (
+	"crypto/tls"
+	"errors"
+	"log"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/blang/vfs"
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/boltdb"
+	"github.com/docker/libkv/store/consul"
+	"github.com/docker/libkv/store/etcd"
+	"github.com/docker/libkv/store/zookeeper"
+	"github.com/hairyhenderson/gomplate/env"
+	consulapi "github.com/hashicorp/consul/api"
+)
+
+// logFatal is defined so log.Fatal calls can be overridden for testing
+var logFatal = log.Fatal
+
+// LibKV -
+type LibKV struct {
+	store store.Store
+	fs    vfs.Filesystem
+}
+
+type SetupDetails struct {
+	sourceType store.Backend
+	client     string
+	options    *store.Config
+}
+
+// New - instantiate a new
+func New(url *url.URL) *LibKV {
+	var s *SetupDetails
+
+	if url.Scheme == "consul" || url.Scheme == "consul+http" {
+		setup, err := setupConsul(url, false)
+		if err != nil {
+			logFatal("consul setup error", err)
+		}
+		s = setup
```

Comment:
Seems like an unnecessary assignment - could just name the variable `setup` on ln39 and remove this (and the rest of the occurrences below